api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -148,4 +148,21 @@ message TcpProxy {
   // The maximum number of unsuccessful connection attempts that will be made before
   // giving up. If the parameter is not specified, 1 connection attempt will be made.
   google.protobuf.UInt32Value max_connect_attempts = 7 [(validate.rules).uint32 = {gte: 1}];
+
+  // Specifies the TCP proxy's hashing policy
+  message HashPolicy {
+    // The source IP will be used to compute the hash used by hash-based load balancing
+    // algorithms.
+    message SourceIp {
+    }
+
+    oneof policy_specifier {
+      option (validate.required) = true;
+
+      SourceIp source_ip = 1;
+    }
+  }
+
+  // Optional configuration for TCP proxy hash policy.
```

Comment:
See my comment below about repeated and maybe limiting the size for now. Do we really need this to be repeated? If so can we make it clear in the docs what happens with all of the repeated policies?

Reply:
Done.